backend/src/controllers/groups-controller.ts

```diff
@@ -0,0 +1,86 @@
+import { Request, Response } from 'express';
+import { v4 as uuid } from 'uuid';
+
+interface Group {
+    name:string;
+    _id:string;
+}
+
+let groups: Group[] = []; //Array of groups
+
+//Creates a group with the desired name and gives a unique ID
+export const createGroup = async (req: Request, res: Response) => {
+    const { name }: { name: string } = req.body;
+
+    //Checks if there's a similar group
+    if (groups.find((group) => group.name === name)) {
+        return res.status(409).json({ error: "The group name was already taken"});
+    }
+    //Validates the provided name
+    if (typeof name !== 'string' || name.length > 30 ){
+        return res.status(400).json({ error: "The group name must not exceed 30 characters"});
+    }
+    //Creates a new group with name and id
+    const newGroup : Group = { name, _id: uuid()};
+    groups.push(newGroup);
```

Comment:
Store the created groups in MongoDB. Use Mongoose + Typescript to create groups. The Group schema is available at `backend/src/models/group.ts`.

Remove the `groups` list upon completion

Reply:
To make alpha-testing easier, DM me your MongoDB username so I can give access to our project's MongoDB database